var/spack/repos/builtin/packages/amrvis/package.py

```diff
@@ -0,0 +1,239 @@
+    conflicts('%pgi',
+              msg='Amrvis currently only builds with gcc and clang')
+    conflicts('%xl',
+              msg='Amrvis currently only builds with gcc and clang')
+    conflicts('%xl_r',
+              msg='Amrvis currently only builds with gcc and clang')
+
+    resource(name='amrex',
+             git='https://github.com/AMReX-Codes/amrex.git',
+             tag='master',
+             placement='amrex')
+
+    def edit(self, spec, prefix):
+        # Set all available makefile options to values we want
+        filter_file(r'^AMREX_HOME\s*=.*',
+                    'AMREX_HOME = {0}'.format('./amrex'),
+                    'GNUmakefile')
+        filter_file(r'^PRECISION\s*=.*',
+                    'PRECISION = {0}'.format(spec.variants['prec'].value),
+                    'GNUmakefile')
+        filter_file(r'^DIM\s*=.*',
+                    'DIM = {0}'.format(spec.variants['dims'].value),
+                    'GNUmakefile')
+        filter_file(r'^PROFILE\s*=.*',
+                    'PROFILE = FALSE',
+                    'GNUmakefile')
+        filter_file(r'^TRACE_PROFILE\s*=.*',
+                    'TRACE_PROFILE = FALSE',
+                    'GNUmakefile')
+        filter_file(r'^COMM_PROFILE\s*=.*',
+                    'COMM_PROFILE = FALSE',
+                    'GNUmakefile')
+        filter_file(r'^COMP\s*=.*',
+                    'COMP = {0}'.format(self.compiler.name),
+                    'GNUmakefile')
+        filter_file(r'^DEBUG\s*=.*',
+                    'DEBUG = {0}'.format(spec.variants['debug'].value).upper(),
+                    'GNUmakefile')
+        filter_file(r'^USE_ARRAYVIEW\s*=.*',
+                    'USE_ARRAY_VIEW = FALSE',
+                    'GNUmakefile')
+        filter_file(r'^USE_MPI\s*=.*',
+                    'USE_MPI = {0}'.format(spec.variants['mpi'].value).upper(),
+                    'GNUmakefile')
+        filter_file(r'^USE_CXX11\s*=.*',
+                    'USE_CXX11 = TRUE',
+                    'GNUmakefile')
+        filter_file(r'^USE_VOLRENDER\s*=.*',
+                    'USE_VOLRENDER = FALSE',
+                    'GNUmakefile')
+        filter_file(r'^USE_PARALLELVOLRENDER\s*=.*',
+                    'USE_PARALLELVOLRENDER = FALSE',
+                    'GNUmakefile')
+        filter_file(r'^USE_PROFPARSER\s*=.*',
+                    'USE_PROFPARSER = FALSE',
+                    'GNUmakefile')
+
+        # Delete all /usr and /opt X library default search paths in makefile
+        filter_file(r'^.*\b(usr|opt)\b.*$',
+                    '# Spack removed INCLUDE_LOCATIONS and LIBRARY_LOCATIONS',
+                    'GNUmakefile')
+
+        # Read GNUmakefile into array
+        with open('GNUmakefile', 'r') as file:
+            contents = file.readlines()
+
+        # Edit GNUmakefile INCLUDES and LIBRARIES to use Spack dependencies.
+        # Assuming the default GNUmakefile doesn't change, this is the best
```

Comment:
This is a big assumption 😄

Reply:
Ugh, I know. Not sure of a more robust thing to do here. I'm not a fan of AMReX's build ideology, but at least they kind of support CMake for that.